common4j/src/main/com/microsoft/identity/common/java/commands/parameters/nativeauth/AcquireTokenNoFixedScopesCommandParameters.java

```diff
@@ -63,17 +63,91 @@ public class AcquireTokenNoFixedScopesCommandParameters extends BaseNativeAuthCo
     @Expose()
     private final Set<String> scopes;
 
+    @Expose()
+    private final String claimsRequestJson;
+
     @Expose()
     @NonNull
     private final AbstractAuthenticationScheme authenticationScheme;
 
+    @Expose()
+    private final String mamEnrollmentId;
+
     @Expose()
     private final boolean forceRefresh;
 
     private final String loginHint;
 
     private final List<Map.Entry<String, String>> extraOptions;
 
+    public Set<String> getScopes() {
+        return this.scopes == null ? null : new HashSet<>(this.scopes);
+    }
+
+    public String getMamEnrollmentId(){
+        return mamEnrollmentId;
+    }
+
+    public void validate() throws ArgumentException {
+        final String methodName = ":validate";
+
+        Logger.verbose(
+                TAG + methodName,
+                "Validating operation params..."
+        );
+
+        // AuthenticationScheme is present...
+        if (null == authenticationScheme) {
+            throw new ArgumentException(
+                    ArgumentException.ACQUIRE_TOKEN_SILENT_OPERATION_NAME, // TODO update argument name
+                    ArgumentException.AUTHENTICATION_SCHEME_ARGUMENT_NAME,
+                    "authentication scheme is undefined"
+            );
+        }
+
+        if (getAccount() == null) {
+            Logger.warn(TAG, "The account set on silent operation parameters is NULL.");
+            // if the authority is B2C, then we do not need check if matches with the account environment
+            // as B2C only exists in one cloud and can use custom domains
+            // This logic should also apply to CIAM authorities
+        }
+    }
+
+    /**
+     * Note - this method may throw a variety of RuntimeException if we cannot perform cloud
+     * discovery to determine the set of cloud aliases.
+     * @return true if the authority matches the cloud environment that the account is homed in.
+     */
+    private boolean authorityMatchesAccountEnvironment() {
```

Comment:
Rather than duplicating, is it possible to re-use the existing method?

Reply:
Is this comment for this method or `performCloudDiscovery` method?